include/envoy/upstream/cluster_manager.h

```diff
@@ -22,6 +22,26 @@
 namespace Envoy {
 namespace Upstream {
 
+/**
+ * ClusterUpdateCallbacks provide a way to exposes Cluster lifecycle event in the
+ * ClusterManager.
+ */
+class ClusterUpdateCallbacks {
+public:
+  virtual ~ClusterUpdateCallbacks() {}
+
+  /**
+   * onClusterAddOrUpdate is called when a new cluster is added or an existing cluster
+   * is updated in the ClusterManager.
```

Comment:
Interfaces in `include/envoy` should use Doxygen docs, include `@param`.

Reply:
Done